### PR TITLE
[1.x] Allow using namespaced aliases as extension names

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

**Comment by stof** [comment]
Actually, can you add a unit test covering this ?

**Comment by nicolas-grekas** [comment]
test added

**Comment by nicolas-grekas** [comment]
Now with some `require_once` so that PHP doesn't cry because of incompatible type hints (because it doesn't know about the aliases yet.)

File: test/Twig/Tests/EnvironmentTest.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-require_once dirname(__FILE__).'/FilesystemHelper.php';
+require_once __DIR__.'/FilesystemHelper.php';
```
Comment (fabpot) [inline test/Twig/Tests/EnvironmentTest.php]: Should be reverted. Even if you're right here, I prefer to not upgrade the code to 5.3+ in 1.x. It has been done in 2.x and I don't want people starting to send random PR about such changes :)

Comment (nicolas-grekas) [inline test/Twig/Tests/EnvironmentTest.php]: reverted

Comment (nicolas-grekas) [inline test/Twig/Tests/EnvironmentTest.php]: haha, who should I fallow? fabpot or fabbot?

**Comment by stof** [comment]
Do we actually need these require_once everywhere ?

**Comment by nicolas-grekas** [comment]
Yes. That's what https://github.com/symfony/symfony/pull/22962 allowed me to spot.

**Comment by nicolas-grekas** [comment]
renamed `Twig\Environment` to `Twig\Twig`

File: test/Twig/Tests/EnvironmentTest.php (hunk 1 above)
Comment (SpacePossum) [inline test/Twig/Tests/EnvironmentTest.php]: (https://github.com/twigphp/Twig/pull/2493 ;) )

File: lib/Twig/Environment.php [hunk 2]
```diff
@@ -758,7 +758,7 @@ public function compileSource($source, $name = null)
 
     public function setLoader(Twig_LoaderInterface $loader)
     {
-        if (!$loader instanceof Twig_SourceContextLoaderInterface && 0 !== strpos(get_class($loader), 'Mock_Twig_LoaderInterface')) {
+        if (!$loader instanceof Twig_SourceContextLoaderInterface && 0 !== strpos(get_class($loader), 'Mock_')) {
```
Comment (nicolas-grekas) [inline lib/Twig/Environment.php]: Mock_Twig_LoaderInterface vs Mock_LoaderInterface when sing the namespaced variant
looking for mock_ is enough IMHO

**Comment by nicolas-grekas** [comment]
Now with aliases that were missing for Twig_Extension_InitRuntimeInterface, Twig_Extension_GlobalsInterface, Twig_SourceContextLoaderInterface and Twig_ExistsLoaderInterface

PR should be ready.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by Koc** [comment]
@fabpot why reverted `\Twig\Twig`?

**Comment by fabpot** [comment]
After trying to use the new class, it felt wrong. `Twig` feels like a singleton, which is wrong. `Environment` conveys the idea that you can have more than one, which is better.

**Pull request merged**